sklearn/utils/estimator_checks.py

```diff
@@ -2416,6 +2417,69 @@ def check_classifiers_regression_target(name, estimator_orig):
         assert_raises_regex(ValueError, msg, e.fit, X, y)
 
 
+@ignore_warnings(category=(DeprecationWarning, FutureWarning))
+def check_estimator_sparse_dense(name, estimator_orig):
+    rng = np.random.RandomState(52)
+    estimator = clone(estimator_orig)
+    estimator_sp = clone(estimator_orig)
+    tags = _safe_tags(estimator_orig)
+    centers = 2 if tags["binary_only"] else None
+    X, y = make_blobs(random_state=rng, cluster_std=0.5, centers=centers)
+    # for put some points to zero to have a little bit of sparsity
+    if tags["requires_positive_X"]:
```

Comment:
This could be updated with `enforce_estimator_tags_X` as soon as #14705 is merged